examples/dmd/parametric_dw_csv.cpp

```diff
@@ -0,0 +1,931 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2013-2022, Lawrence Livermore National Security, LLC
+ * and other libROM project developers. See the top-level COPYRIGHT
+ * file for details.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT)
+ *
+ *****************************************************************************/
+
+// Compile with: make parametric_tw_csv
+//
+// Generate CSV databse on heat conduction with: heat_conduction_csv.sh
+//
+// =================================================================================
+//
+// Parametric serial DMD command:
+//   mpirun -np 8 parametric_tw_csv -o parametric_csv_serial -rdim 16 -dtc 0.01
+//
+// Final-time prediction error (Last line in run/parametric_csv_serial/hc_par5_prediction_error.csv):
+//   0.0012598331433506
+//
+// Parametric time windowing DMD command:
+//   mpirun -np 8 parametric_tw_csv -o parametric_csv_tw -rdim 16 -nwinsamp 25 -dtc 0.01
+//
+// Final-time prediction error (Last line in run/parametric_csv_tw/hc_par5_prediction_error.csv):
+//   0.0007348122935743
+//
+// =================================================================================
+//
+// Description: Parametric solution manifold decomposition DMD on general CSV datasets.
+//
+// User specify file locations and names by -list LIST_DIR -train-set TRAIN_LIST -test-set TEST_LIST -data DATA_DIR -var VAR_NAME -o OUT_DIR
+//
+// File structure:
+//  1. LIST_DIR/TRAIN_LIST.csv             -- each row specifies one training DATASET
+//  2. LIST_DIR/TEST_LIST.csv              -- each row specifies one testing DATASET
+//  3. LIST_DIR/DATASET.csv                -- each row specifies one STATE in DATASET
+//  4. DATA_DIR/dim.csv                    -- specifies the dimension of VAR_NAME
+//  5. DATA_DIR/DATASET/tval.csv           -- specifies the time instances
+//  6. DATA_DIR/DATASET/STATE/VAR_NAME.csv -- each row specifies one value of VAR_NAME of STATE
+//  7. DATA_DIR/SPATIAL_IDX.csv            -- (optional) each row specifies one spatial index of VAR_NAME
+//  8. DATA_DIR/DATASET/TEMPORAL_IDX.csv   -- (optional) specifies the first and last temporal index in DATASET
+//  9. run/OUT_DIR/INDICATOR_IDX.csv       -- (optional) each row specifies one indicator index
+// 10. run/OUT_DIR/INDICATOR_VAL.csv       -- (optional) each row specifies one indicator endpoint value
+
+#include "mfem.hpp"
```

Comment:
Can dependency on mfem be removed in this file and others that are purely data driven? These examples build only with the option `compile.sh -m`. If they are independent of mfem, the build system could compile them also without the `-m` flag.

Reply:
Removing the dependency on mfem can be done in a future PR, but let's keep it in mind.

Reply:
I have added that in the description of this PR.